src/string-formatter.ts

```diff
@@ -0,0 +1,44 @@
+import {format} from "util";
+
+export class StringFormatter {
+  public format(parameters: any[]): string {
+    const [, ...textParams] = parameters;
+    let [textFormat] = parameters;
+    let text;
+
+    try {
+      // Our sanitizer gives better results, try with it first
+      textFormat = this.toString(textFormat);
+    } catch {
+      // Fallback to `format`
+      textFormat = format(textFormat);
+    }
+
+    if (textParams.length > 0) {
+      text = format(textFormat, ...textParams);
+    } else {
+      text = textFormat;
+    }
+
+    return text;
+  }
+
+  private toString(value: any): string {
```

Comment:
This probably can be changed to `unknown`.

```suggestion
  private toString(value: unknown): string {
```